protocol/binary_test.go

```diff
@@ -452,16 +496,33 @@ func TestMapEOFFailure(t *testing.T) {
 
 func TestSet(t *testing.T) {
 	tests := []encodeDecodeTest{
-		{vset(wire.TBool), []byte{0x02, 0x00, 0x00, 0x00, 0x00}},
+		{"small set", vset(wire.TBool), []byte{0x02, 0x00, 0x00, 0x00, 0x00}},
 		{
-			vset(wire.TBool, vbool(true), vbool(false), vbool(true)),
+			"large set", vset(wire.TBool, vbool(true), vbool(false), vbool(true)),
 			[]byte{0x02, 0x00, 0x00, 0x00, 0x03, 0x01, 0x00, 0x01},
 		},
 	}
 
 	checkEncodeDecode(t, wire.TSet, tests)
 }
 
+func TestSetBeginEncode(t *testing.T) {
+	var streamBuff bytes.Buffer
+	var err error
```

Comment:
optional nit: can group
```suggestion
	var (
		streamBuff bytes.Buffer
		err error
	)
```